packages/engine/src/avatar/components/AvatarAnimationComponent.ts

```diff
@@ -103,58 +121,37 @@ export const AvatarRigComponent = defineComponent({
 
   reactor: function () {
     const entity = useEntityContext()
-    const debugEnabled = useHookstate(getMutableState(RendererState).avatarDebug)
     const rigComponent = useComponent(entity, AvatarRigComponent)
-    const pending = useOptionalComponent(entity, AvatarPendingComponent)
-    const visible = useOptionalComponent(entity, VisibleComponent)
-    const modelComponent = useOptionalComponent(entity, ModelComponent)
+    const gltfComponent = useOptionalComponent(entity, GLTFComponent)
     const locomotionAnimationState = useHookstate(
       getMutableState(AnimationState).loadedAnimations[preloadedAnimations.locomotion]
     )
 
+    /** @todo move asset loading to a new VRMComponent */
     useEffect(() => {
-      if (!visible?.value || !debugEnabled.value || pending?.value || !rigComponent.value.normalizedRig?.hips?.node)
-        return
-
-      const helper = new SkeletonHelper(rigComponent.value.vrm.scene as Group)
-      helper.frustumCulled = false
-      helper.name = `target-rig-helper-${entity}`
-
-      const helperEntity = createEntity()
-      setVisibleComponent(helperEntity, true)
-      addObjectToGroup(helperEntity, helper)
-      setComponent(helperEntity, NameComponent, helper.name)
-      setObjectLayers(helper, ObjectLayers.AvatarHelper)
-
-      setComponent(helperEntity, ComputedTransformComponent, {
-        referenceEntities: [entity],
-        computeFunction: () => {
-          // this updates the bone helper lines
-          helper.updateMatrixWorld(true)
-        }
-      })
-
+      if (!rigComponent?.avatarURL?.value) return
+      setComponent(entity, GLTFComponent, { src: rigComponent.avatarURL.value })
       return () => {
-        removeEntity(helperEntity)
+        removeComponent(entity, GLTFComponent)
       }
-    }, [visible, debugEnabled, pending, rigComponent.normalizedRig])
+    }, [rigComponent?.avatarURL?.value])
```

Comment:
`avatarURL` is a redundant field, we can just set the `GLTFComponent` src directly